elastic/kibana
# [Ingest Node Pipelines] Refactor pipeline simulator code

This PR is prep work for the enhancements planned for the “Test pipeline” workflow.

Changes include:
- Move test pipeline flyout code inside the `pipeline_processors_editor` directory
- Only sending `processors` and `on_failure` processors to the simulate API; I don't see any benefit to sending the other pipeline fields (version, description). 
- Export a single context from the processors editor

Future polish
- Consider formatting simulate error message as in https://github.com/elastic/kibana/pull/70167
- There is now the issue of closing the flyout once the editor flyout is open. I think we might be able to resolve this by using the new global flyout component.

### How to test
No functionality should have changed here. You can follow the instructions on https://github.com/elastic/kibana/pull/64223 to verify there are no regressions.
